### PR TITLE
rust: bump to 1.47 stable

### PR DESCRIPTION
**Pull request by benma**
And use updated Docker image based on this.

I tried this to fix the reproducible builds issue, but it didn't help. Still a valid PR :P

### CODE REVIEW

File: .ci/run-container-ci [hunk 1]
```diff
@@ -25,7 +25,7 @@
 set -e
 set -x
 
-CONTAINER=shiftcrypto/firmware_v2:18
+CONTAINER=shiftcrypto/firmware_v2:19
```
Comment (x1ddos) [inline .ci/run-container-ci:28]: Do the same in dockerenv? https://github.com/digitalbitbox/bitbox02-firmware/blob/ddcdb0e/scripts/dockerenv.sh#L27.

There's also `docker-compose.yml` in the root of the repo and it references unversioned image. Is that compose config file even used?

Comment (benma) [reply .ci/run-container-ci:28]: dockerenv is deliberately at unversioned, so it works after `make dockerinit` and `docker pull shiftcrypto/firmware_v2`.

I never used docker-compose.yml, so let's delete it.

Comment (x1ddos) [reply .ci/run-container-ci:28]: :+1:

**Review by x1ddos — APPROVED** [review]
LGTM

**Pull request merged**